### PR TITLE
[RHPAM-3654] - Wrong archetype used to generate project from provided…

### PR DESCRIPTION
**Pull request by spolti**
… asset

Signed-off-by: spolti <fspolti@redhat.com>

Many thanks for submitting your Pull Request :heart:! 

Please make sure your PR meets the following requirements:

- [x] You have read the [contributors guide](README.md#contributing-to-kogito-images-repository)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [x] Your feature/bug fix has a testcase that verifies it
- [ ] You've tested the new feature/bug fix in an actual OpenShift cluster

### CODE REVIEW

**Comment by spolti** [comment]
will cherry-pick to 1.5.x once tests passes and we have the approvals.

File: tests/features/kogito-builder.feature [hunk 1]
```diff
@@ -102,3 +102,22 @@ Feature: kogito-builder image tests
...
+
+#### SpringBoot Scenarios
+
+  Scenario: Verify that the Kogito Maven archetype is generating the project and compiling it correctly when runtime is springboot
+    Given s2i build /tmp/kogito-examples from dmn-example using nightly-master and runtime-image quay.io/kiegroup/kogito-runtime-jvm:latest
+      | variable       | value          |
+      | KOGITO_VERSION | 2.0.0-SNAPSHOT |
+      | RUNTIME_TYPE   | springboot     |
+    Then file /home/kogito/bin/project-1.0-SNAPSHOT.jar should exist
+    And s2i build log should contain Generating quarkus project structure using the kogito-springboot-archetype archetype...
```
Comment (radtriste) [inline tests/features/kogito-builder.feature]:
```suggestion
    And s2i build log should contain Generating springboot project structure using the kogito-springboot-archetype archetype...
```

File: tests/features/rhpam-kogito-builder-jvm.feature [hunk 2]
```diff
@@ -54,6 +54,25 @@ Feature: rhpam-kogito-builder-rhel8 feature.
...
+      | request_method  | POST                                                                                             |
+      | content_type    | application/json                                                                                 |
+      | request_body    | {"Driver": {"Points": 2}, "Violation": {"Type": "speed","Actual Speed": 120,"Speed Limit": 100}} |
+
+#### SpringBoot Scenarios
+
+  Scenario: Verify that the Kogito Maven archetype is generating the project and compiling it correctly when runtime is springboot
+    Given s2i build /tmp/kogito-examples from dmn-example using master and runtime-image rhpam-7/rhpam-kogito-runtime-jvm-rhel8:latest
+      | variable       | value          |
+      | KOGITO_VERSION | 2.0.0-SNAPSHOT |
```
Comment (radtriste) [inline tests/features/rhpam-kogito-builder-jvm.feature:72]:
```suggestion
      | KOGITO_VERSION | 1.5.0.redhat-00001 |
```

**Review by radtriste — CHANGES_REQUESTED** [review]

**Comment by radtriste** [comment]
The other failing test: `Verify that the Kogito Quarkus Serverless Workflow Extension is building the service properly`
should be solved by https://github.com/kiegroup/kogito-images/pull/507 (please review that one)

File: tests/features/rhpam-kogito-builder-jvm.feature [hunk 3]
```diff
@@ -54,6 +54,25 @@ Feature: rhpam-kogito-builder-rhel8 feature.
...
+      | wait            | 80                                                                                               |
+      | expected_phrase | Should the driver be suspended?                                                                  |
+      | request_method  | POST                                                                                             |
+      | content_type    | application/json                                                                                 |
+      | request_body    | {"Driver": {"Points": 2}, "Violation": {"Type": "speed","Actual Speed": 120,"Speed Limit": 100}} |
+
+#### SpringBoot Scenarios
+
+  Scenario: Verify that the Kogito Maven archetype is generating the project and compiling it correctly when runtime is springboot
+    Given s2i build /tmp/kogito-examples from dmn-example using master and runtime-image rhpam-7/rhpam-kogito-runtime-jvm-rhel8:latest
```
Comment (radtriste) [inline tests/features/rhpam-kogito-builder-jvm.feature]:
```suggestion
    Given s2i build /tmp/kogito-examples from dmn-example using nightly-master and runtime-image rhpam-7/rhpam-kogito-runtime-jvm-rhel8:latest
```

File: tests/features/rhpam-kogito-builder-jvm.feature (hunk 2 above)
Comment (spolti) [reply tests/features/rhpam-kogito-builder-jvm.feature:72]: this is main branch, shouldn't it be used only for 1.5.x branch?

Comment (radtriste) [reply tests/features/rhpam-kogito-builder-jvm.feature:72]: Then I guess you need also to update other test keeping 2.0.0-SNAPSHOT

File: tests/features/rhpam-kogito-builder-jvm.feature [hunk 4]
```diff
@@ -62,4 +62,23 @@ Feature: rhpam-kogito-builder-rhel8 feature.
...
+      | request_body    | {"Driver": {"Points": 2}, "Violation": {"Type": "speed","Actual Speed": 120,"Speed Limit": 100}} |
+
+#### SpringBoot Scenarios
+
+  Scenario: Verify that the Kogito Maven archetype is generating the project and compiling it correctly when runtime is springboot
+    Given s2i build /tmp/kogito-examples from dmn-example using nightly-master and runtime-image rhpam-7/rhpam-kogito-runtime-jvm-rhel8:latest
+      | variable       | value          |
+      | KOGITO_VERSION | 2.0.0-SNAPSHOT |
+      | RUNTIME_TYPE   | springboot     |
+    Then file /home/kogito/bin/project-1.0-SNAPSHOT.jar should exist
```
Comment (radtriste) [inline tests/features/rhpam-kogito-builder-jvm.feature:74]: I guess it should be `/home/kogito/bin/quarkus-run.jar` or ?

Comment (spolti) [reply tests/features/rhpam-kogito-builder-jvm.feature:74]: i don't think so, it is springboot test.

Comment (radtriste) [reply tests/features/rhpam-kogito-builder-jvm.feature:74]: ah true sorry ...

**Review by radtriste — APPROVED** [review]

**Comment by radtriste** [comment]
@spolti to you for cherry-picking ;)

**Comment by spolti** [comment]
@radtriste thanks :)

**Pull request merged**